src/FakeItEasy/ArgumentConstraintManagerExtensions.cs

```diff
@@ -271,6 +272,23 @@ public static T IsEqualTo<T>(this IArgumentConstraintManager<T> manager, T value
                 x => x.Write("equal to ").WriteArgumentValue(value));
         }
 
+        /// <summary>
+        /// Tests that the passed in argument is equal to the specified value using provided equality comparer.
+        /// </summary>
+        /// <typeparam name="T">The type of the argument.</typeparam>
+        /// <param name="manager">The constraint manager to match the constraint.</param>
+        /// <param name="value">The value to compare to.</param>
+        /// <param name="comparer">The comparer to use for equality comparison.</param>
+        /// <returns>A dummy argument value.</returns>
+        public static T IsEqualTo<T>(this IArgumentConstraintManager<T> manager, T value, IEqualityComparer<T> comparer)
+        {
+            Guard.AgainstNull(manager, nameof(manager));

```

Comment:
There's a concern here that `comparer` will be `null`. I would `Guard.Against` it here and also add a test to `EqualToWithComparerConstraintTests` to ensure that it's null-guarded. You can look for examples of `BeNullGuarded` elsewhere in the code.

Reply:
I had only one concern with the test for guarding against comparer being null, I was not sure if it is the correct you would write such test for it, so have a closer look to that :)